proselint/checks/__init__.py

```diff
@@ -1 +1,394 @@
+    # req non-text character around
+    # -> finds item as long it is surrounded by any non-word character:
+    #       - whitespace
+    #       - punctuation
+    #       - newline ...
+
+
+def existence_check(  # noqa: PLR0913, PLR0917
+    text: str,
+    re_items: list,
+    err: str,
+    msg: str,
+    ignore_case: bool = True,
+    string: bool = False,  # todo: why not default on?
+    offset: int = 0,  # todo: some checks set this strangely
+    padding: Pd = Pd.sep_in_txt,
+    dotall: bool = False,
+    excluded_topics: Optional[list] = None,
+    exceptions=(),
+) -> list[ResultCheck]:
+    """Build a checker that prohibits certain words or phrases."""
+    flags = 0
+    if ignore_case:
+        flags |= re.IGNORECASE
+    if string:
+        flags |= re.UNICODE
+    if dotall:
+        flags |= re.DOTALL
+
+    errors: list[ResultCheck] = []
+
+    # If the topic of the text is in the excluded list, return immediately.
+    if excluded_topics:
+        tps = topics(text)
+        if any(t in excluded_topics for t in tps):
+            return errors
+
+    rx = "|".join(padding.format(_item) for _item in re_items)
+    for m in re.finditer(rx, text, flags=flags):
+        txt = m.group(0).strip()
+        if any(re.search(exception, txt) for exception in exceptions):
+            continue
+        errors.append(
+            (m.start() + 1 + offset, m.end() + offset, err, msg.format(txt), None),
+        )
+        # TODO: doesn't the padding alter the start+end?
+
+    return errors
+
+
+def simple_existence_check(
+    text: str, pattern: str, err: str, msg: str, offset: int = 0
+):
+    """Build a checker for single patters.
+    in comparison to existence_check:
+        - does not work on lists
+        - no padding
+        - excluded topics or exceptions
+
+        TODO: maybe add ignorecase
+    """
+
+    return [
+        (inst.start() + offset, inst.end() + offset, err, msg, None)
+        for inst in re.finditer(pattern, text)
+    ]
+
+
+def detector_50_Cent(text: str) -> tuple[str, float]:
+    """Determine whether 50 Cent is a topic."""
+    keywords = [
+        "50 Cent",
+        "rap",
+        "hip hop",
+        "Curtis James Jackson III",
+        "Curtis Jackson",
+        "Eminem",
+        "Dre",
+        "Get Rich or Die Tryin'",
+        "G-Unit",
+        "Street King Immortal",
+        "In da Club",
+        "Interscope",
+    ]
+    num_keywords = sum(word in text for word in keywords)
+    return "50 Cent", float(num_keywords > 2)
+
+
+def topics(text: str) -> list[str]:
+    """Return a list of topics."""
+    detectors = [
+        detector_50_Cent,
+    ]
+    ts = [detector(text) for detector in detectors]
+    return [t[0] for t in ts if t[1] > 0.95]
+
+
+def context(text, position, level="paragraph"):
+    """Get sentence or paragraph that surrounds the given position."""
+    if level == "sentence":
+        pass
+    elif level == "paragraph":
+        pass
+
+    return ""
```

Comment:
Has this function always passed? I can't remember if it was previously functional, or in the process of being worked out.

Reply:
yes, this is a todo from someone else. i kept it